### ISSUE TITLE
`Show tab search button` toggle doesn't work when in vertical-tabs mode

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#32919]
## Description 

`Show tab search button` toggle doesn't work when in vertical-tabs mode

Sadly, found while testing https://github.com/brave/brave-browser/issues/32809 (maybe/hopefully this is macOS-only?)

## Steps to Reproduce

1. install `1.60.11`
2. launch Brave
3. context-click on any tab
4. choose `Use Vertical Tabs`
5. open `brave://settings/appearance`
6. click to (try to) toggle `Show tab search button`

## Actual result:

Sometimes, nothing seemingly happens; other times, the toggle appears to work but doesn't

![show-tab-search](https://github.com/brave/brave-browser/assets/387249/fd571d44-6e5c-4937-a3af-bef1f79ec7c0)

## Reproduces how often: 

100%

## Brave version (brave://version info)

```
Brave | 1.60.11 Chromium: 117.0.5938.48 (Official Build) nightly (x86_64)
-- | --
Revision | 31e5a6923b9af06a63afc70bd2907933cfd26df7
OS | macOS Version 11.7.9 (Build 20G1426)
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @sangwoo108 @fallaciousreasoning @rebron @brave/qa-team

**Comment by stephendonner** [comment]
Also reproduces with

```
Brave | 1.58.120 Chromium: 117.0.5938.48 (Official Build) (x86_64)
-- | --
Revision | 8b2e2785c90c076fc1413acb2c003e603eac02d3
OS | macOS Version 11.7.9 (Build 20G1426)
```

**Comment by stephendonner** [comment]
Yeah, this appears to be macOS-only (phew), based on my initial testing of `1.60.11` on `Windows 10`, `x64`.

**Comment by MadhaviSeelam** [comment]
Reproduced the issue in Linux in `1.58.120 Chromium: 117.0.5938.48`.  Tab search button is shown and the Search bubble is shown even when Show tab search button is disabled

![image (1)](https://github.com/brave/brave-browser/assets/98358127/4621f46f-0256-4a95-a028-8586e3e07f9a)

**Comment by sangwoo108** [comment]
https://bravesoftware.slack.com/archives/C01EVLWS8R5/p1694466274375489

Still we haven't come to the conclusion if we should apply the pref to the vertical tab strip too.
